src/components/AllArticles.jsx

```diff
@@ -1,48 +1,54 @@
-
-
+import { Link } from "react-router-dom";
 import React, { useEffect, useState } from "react";
-import  {getArticles} from "../api";
+import { getArticles } from "../api";
 import LoadingSpinner from "../loading";
 
-
-export const AllArticles = () => {  
-const [isLoading, setIsLoading] = useState(false);
+export const AllArticles = () => {
+  const [isLoading, setIsLoading] = useState(false);
   const [articles, setArticles] = useState([]);
 
   useEffect(() => {
-  setIsLoading(true)
-  getArticles().then((articles) => {
-    setIsLoading(false)
-    console.log(articles);
-setArticles(articles)
-
-  }).catch((err) => {
-      setIsLoading(false);
-      console.log(err);
-    })
-  },[])
-
-return (
-  <div>
-    {isLoading ? <LoadingSpinner /> : 
-    <ul>
-      {articles.map((article) => {
-        return (
-       <div>
-        <li key={article.article_id}>
-
-          <h3>Title - {article.title}</h3>
-          <h4>Author - {article.author}</h4>
-          <p>Topic - {article.topic}</p>
-          <hr></hr>
-        </li>
-       </div>
+    setIsLoading(true);
+    getArticles()
+      .then((articles) => {
+        setIsLoading(false);
+        setArticles(articles);
+      })
+      .catch((err) => {
+        setIsLoading(false);
+        console.log(err);
+      });
+  }, []);
+
+  return (
+    <div>
+      <h2>Articles</h2>
+      <fieldset>
+        <legend>Topics</legend>
+        <Link to="/articles">All</Link>
+        <Link to="/articles/coding">Coding</Link>
```

Comment:
Still some static links here for coding, cooking, football. think about using the same sort of logic you used to handle this in your app.js here too.